### PR TITLE
Support Reader Listener

### PR DESCRIPTION
**Pull request by k2la**
Add listener function to the Reader, using the C++ client listener support.
- Add listener for reader
- Add example and e2e test of using the reader listener

### CODE REVIEW

File: src/Reader.cc [hunk 1]
```diff
@@ -66,33 +105,41 @@ class ReaderNewInstanceWorker : public Napi::AsyncWorker {
...
-                                    this->readerConfig->GetCReaderConfig(), &(this->cReader));
-    delete this->readerConfig;
+                                    this->readerConfig->GetCReaderConfig(), &this->readerWrapper->cReader);
     if (result != pulsar_result_Ok) {
       SetError(std::string("Failed to create reader: ") + pulsar_result_str(result));
       return;
+    } else {
+      this->listener = this->readerConfig->GetListenerCallback();
     }
+    delete this->readerConfig;
```
Comment (hrsakai) [inline src/Reader.cc:115]: if `result` is not ok, won't this line be executed?

File: src/Reader.cc [hunk 2]
```diff
@@ -176,13 +228,31 @@ class ReaderCloseWorker : public Napi::AsyncWorker {
...
+  this->Unref();
+  this->listener->callback.Release();
+  this->listener = nullptr;
+}
+
+Reader::~Reader() {
+  if (this->listener) {
+    this->CleanupListener();
+  }
+  pulsar_reader_free(this->wrapper->cReader);
```
Comment (hrsakai) [inline src/Reader.cc:257]: Can we delete this line because `pulsar_reader_free` is executed on `CReaderWrapper::~CReaderWrapper`?

Comment (k2la) [reply src/Reader.cc]: OK. I removed it.

Comment (k2la) [reply src/Reader.cc]: Exactly. If the result is not ok, this is not be executed.
I fixed that this is be done if the result is not ok.

File: src/Reader.cc (hunk 1 above)
Comment (k2la) [reply src/Reader.cc:116]: Exactly. If the result is not ok, this is not be executed.
I fixed that this is be done if the result is not ok.

Comment (hrsakai) [reply src/Reader.cc]: I think it is better to write as follows.
https://github.com/apache/pulsar-client-node/blob/v1.3.0/src/Consumer.cc#L180-L187

Comment (k2la) [reply src/Reader.cc]: I implemented `createReaderCallback` like [subscribeCallback](https://github.com/apache/pulsar-client-node/blob/v1.3.0/src/Consumer.cc#L178-L190).

**Review by hrsakai — APPROVED** [review]
LGTM

**Pull request merged**